src/atomic64/nightly.rs

```diff
@@ -0,0 +1,86 @@
+// Copyright 2014 The Prometheus Authors
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::f64;
+use std::sync::atomic::{AtomicI64 as StdAtomicI64, AtomicU64 as StdAtomicU64, Ordering};
+
+pub struct AtomicF64 {
+    inner: StdAtomicU64,
+}
+
+fn u64_to_f64(val: u64) -> f64 {
+    f64::from_bits(val)
+}
+
+fn f64_to_u64(val: f64) -> u64 {
+    f64::to_bits(val)
+}
+
+impl super::Atomic<f64> for AtomicF64 {
+    fn new(val: f64) -> AtomicF64 {
+        AtomicF64 {
+            inner: StdAtomicU64::new(f64_to_u64(val)),
+        }
+    }
+
+    #[inline]
+    fn set(&self, val: f64) {
+        self.inner.store(f64_to_u64(val), Ordering::Relaxed);
+    }
+
+    #[inline]
+    fn get(&self) -> f64 {
+        u64_to_f64(self.inner.load(Ordering::Relaxed))
+    }
+
+    #[inline]
+    fn inc_by(&self, delta: f64) {
+        loop {
+            let current = self.inner.load(Ordering::Acquire);
+            let new = u64_to_f64(current) + delta;
+            let swapped = self.inner
+                .compare_and_swap(current, f64_to_u64(new), Ordering::Release);
+            if swapped == current {
+                return;
+            }
+        }
+    }
+}
+
+pub struct AtomicI64 {
+    inner: StdAtomicI64,
+}
+
+impl super::Atomic<i64> for AtomicI64 {
+    fn new(val: i64) -> AtomicI64 {
+        AtomicI64 {
+            inner: StdAtomicI64::new(val),
+        }
+    }
+
+    #[inline]
+    fn set(&self, val: i64) {
+        self.inner.store(val, Ordering::Relaxed);
+    }
+
+    #[inline]
+    fn get(&self) -> i64 {
+        self.inner.load(Ordering::Acquire)
```

Comment:
do we need to use Acquire here?

Reply:
I just keep it as it is here and normally it is acquire-release. For x86 indeed it is the lowest factual order.